### PR TITLE
Xl primarybutton

### LINKED ISSUE #7: Primary Button Component
## Description

Create a button component according to [Figma Design](https://www.figma.com/file/PFCGAeZzwLEyzNaovh6Zfq/Toriis?node-id=179%3A270&t=dNLdQczcKAhNGwaS-0)

image

## Learning Objectives

- Tailwind CSS
- React Components

## Todo

These points are a rough guideline. Please feel free to discuss with others on the team about the best way to design this component!

- [ ] Create a `PrimaryButton.tsx` component in [components](https://github.com/toriis-portal/toriis/tree/main/src/components) folder. Notice that the button should:  
  - Handle click animation 
  - Handle page redirections
- [ ] Style the button so that it matches designs 
  - Check out [Box Shadow](https://tailwindcss.com/docs/box-shadow) and [Pointer Events](https://tailwindcss.com/docs/pointer-events) in Tailwind CSS
  - Note that we are using [Hero Icons](https://heroicons.com/)

## Useful Links

- [Figma Design](https://www.figma.com/file/PFCGAeZzwLEyzNaovh6Zfq/Toriis?node-id=179%3A270&t=dNLdQczcKAhNGwaS-0)
- [Hero Icons](https://heroicons.com/)

## Timeline 

By start of Thursday dev night

### PR DESCRIPTION
**Pull request by 21xiaofanli**
## Summary
Created an animated button for the links "More About Fossil Fuel" and "Learn About Investments"
Closes #7

## Changes

- Added a PrimaryButton file that defines the component

## Screenshots

image

## Requests / Responses

`userRouter.getUser.useQuery({id: string})` returns a list of users

```
{
  "data": {
    "type": "users",
    "id": "4",
    "attributes": {
      "name": "The Dude",
      "email": "thedudeabides@wee.net",
      "last-logged-in": null,
      "created-at": "2016-10-20T17:45:08.190Z",
      "updated-at": "2016-10-20T17:45:08.190Z"
    },
    "links": {
      "self": "/users/4"
    }
  }
}
```

### CODE REVIEW

File: src/components/PrimaryButton.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { ArrowRightIcon } from '@heroicons/react/24/solid'
+import Link from 'next/link'
+
+const PrimaryButton: React.FC<{ text: string; link: string }> = ({
```
Comment (zorazrr) [inline src/components/PrimaryButton.tsx:4]: We can extract what is in `FC<{text:string; link: string}>` into an interface 
```
buttonProps {
   text: string,
   link: string
}
```

File: src/components/PrimaryButton.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  text,
+  link,
+}) => {
+  return (
+    <Link href={link}>
+      <button className="rounded border-2 border-solid border-cobalt bg-lightBlue px-5 py-1 shadow-[-8px_8px_0px_0px] shadow-cobalt hover:shadow-[-5px_5px_0px_0px] hover:shadow-cobalt ">
+        <b className="font-semibold">
+          {text} <ArrowRightIcon className="bold inline h-9 w-5" />{' '}
```
Comment (zorazrr) [inline src/components/PrimaryButton.tsx:12]:
```suggestion
          {text} <ArrowRightIcon className="bold inline h-9 w-5" />
```

**Review by zorazrr — CHANGES_REQUESTED** [review]
The button is looking good! Just need to address some small fixes. Additionally, `package-lock.json` should not be here as we are using yarn not npm.

File: src/components/PrimaryButton.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { ArrowRightIcon } from '@heroicons/react/24/solid'
+import Link from 'next/link'
+
+const PrimaryButton: React.FC<{ text: string; link: string }> = ({
+  text,
+  link,
+}) => {
+  return (
+    <Link href={link}>
+      <button className="rounded border-2 border-solid border-cobalt bg-lightBlue px-5 py-1 shadow-[-8px_8px_0px_0px] shadow-cobalt hover:shadow-[-5px_5px_0px_0px] hover:shadow-cobalt ">
```
Comment (zorazrr) [inline src/components/PrimaryButton.tsx:10]: This line is super long. I added a new package `cslx` that will help with this specific situation with Tailwind CSS. An example usage can be found [here](https://dev.to/joserfelix/6-tips-for-tailwind-css-development-with-resources-33i4#:~:text=My%20solution%20to%20this%20type%20of%20long%20class%20names%20is%20using%20comma%2Dseparated%20values%20in%20clsx.%20Like%20this%3A)! So merge main and try it out!

**Comment by 21xiaofanli** [comment]
Finished the review

**Review by zorazrr — APPROVED** [review]

File: src/components/Buttons/PrimaryButton.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { ArrowRightIcon } from '@heroicons/react/24/solid'
+import Link from 'next/link'
+import clsx from 'clsx'
+
+interface ButtonProps {
+  text: string
+  link: string
+}
+
+const PrimaryButton = ({ text, link }: ButtonProps) => {
```
Comment (zorazrr) [inline src/components/Buttons/PrimaryButton.tsx:10]: One last thing, for consistency, we want to maintain the same type def format across components. Check out [NavButton](https://github.com/toriis-portal/toriis/blob/main/src/components/NavButton.tsx) for instance. We'd want to do something like `const PrimaryButton: FC<ButtonProps> = ({ text, link }) => {` here.

**Pull request merged**